go/types/list.go

```diff
@@ -185,18 +219,22 @@ func (l List) IterAll(f listIterAllFunc) {
 	})
 }
 
+// Iterator returns a ListIterator which can be used to iterate efficiently over a list.
 func (l List) Iterator() ListIterator {
 	return l.IteratorAt(0)
 }
 
+// IteratorAt returns a ListIterator starting at index.
 func (l List) IteratorAt(index uint64) ListIterator {
 	return ListIterator{newCursorAtIndex(l.seq, index)}
 }
 
+// Diff returns the diff of two different lists.
```

Comment:
Diff streams the diff from last to l to changes. Caller can close closeChan to cancel the diff operation.